src/Atmos/Model/ref_state.jl

```diff
@@ -95,3 +108,63 @@ function atmos_init_aux!(
     e_pot = gravitational_potential(atmos.orientation, aux)
     aux.ref_state.ρe = ρ * total_energy(e_kin, e_pot, ts)
 end
+
+using ..MPIStateArrays: vars
+using ..DGMethods: init_ode_state
+using ..DGMethods.NumericalFluxes:
+    CentralNumericalFluxFirstOrder,
+    CentralNumericalFluxSecondOrder,
+    CentralNumericalFluxGradient
+
+struct PressureGradientModel <: BalanceLaw end
+vars_state_auxiliary(::PressureGradientModel, T) = @vars(p::T)
+vars_state_conservative(::PressureGradientModel, T) = @vars(∇p::SVector{3, T})
+vars_state_gradient(::PressureGradientModel, T) = @vars()
+vars_state_gradient_flux(::PressureGradientModel, T) = @vars()
+function init_state_auxiliary!(
+    m::PressureGradientModel,
+    state_auxiliary::MPIStateArray,
+    grid,
+) end
+function init_state_conservative!(
+    ::PressureGradientModel,
+    state::Vars,
+    aux::Vars,
+    coord,
+    t,
+) end
+function flux_first_order!(
+    ::PressureGradientModel,
+    flux::Grad,
+    state::Vars,
+    auxstate::Vars,
+    t::Real,
+    direction,
+)
+    flux.∇p -= auxstate.p * I
+end
+flux_second_order!(::PressureGradientModel, _...) = nothing
+source!(::PressureGradientModel, _...) = nothing
+boundary_state!(nf, ::PressureGradientModel, _...) = nothing
+
+∇reference_pressure(::NoReferenceState, state_auxiliary, grid) = nothing
+function ∇reference_pressure(::ReferenceState, state_auxiliary, grid)
+    FT = eltype(state_auxiliary)
+    ∇p = similar(state_auxiliary; vars = @vars(∇p::SVector{3, FT}))
+
+    grad_model = PressureGradientModel()
+    grad_dg = DGModel(
+        grad_model,
+        grid,
+        CentralNumericalFluxFirstOrder(),
```

Comment:
Let's make sure that the numerical flux with which the reference state pressure gradient is computed is the same as that which is used in the momentum flux computation. @tapios made the point below (point 3 in the list).

Reply:
The choice of numerical fluxes doesn't matter for taking the reference state pressure gradient since it is a continuous field. I added a comment that says so. Do we need to handle reference states with discontinuous pressure ?